addons/account/tests/test_tax.py

```diff
@@ -92,29 +162,46 @@ def compare_values(results, expected_values, rounding):
-        is_round_globally = evaluation_context_kwargs.get('rounding_method') == 'round_globally'
-        rounding = 0.000001 if is_round_globally else 0.01
-        compare_values(results, expected_values, rounding)
-
-        # Check the reverse in case of round_globally.
-        if is_round_globally:
-            evaluation_context = taxes._eval_taxes_computation_prepare_context(results['total_excluded'], quantity, **{
-                **evaluation_context_kwargs,
-                'reverse': True,
-            })
-            results = taxes._eval_taxes_computation(taxes_computation, evaluation_context)
-            compare_values(results, expected_values, rounding)
-            delta = sum(x['tax_amount_factorized'] for x in results['tax_values_list'] if x['price_include'])
-            self.assertEqual(
-                float_round(results['total_excluded'] + delta, precision_rounding=rounding),
-                float_round(price_unit, precision_rounding=rounding),
-            )
+        params = test['params']
+        if params['test'] == 'taxes_computation':
+            is_round_globally = params['is_round_globally']
+            rounding = 0.000001 if is_round_globally else 0.01
+
+            for results_key in ('py_results', 'js_results'):
+                with self.subTest(results_key=results_key):
+                    if results_key not in test:
+                        continue
+
+                    results = test[results_key]
+                    compare_taxes_computation_values(results['results'], test['expected_values'], rounding)
+
+                    # Check the reverse in case of round_globally.
+                    if is_round_globally:
+                        compare_taxes_computation_values(results['reverse_results'], test['expected_values'], rounding)
+                        delta = sum(x['tax_amount_factorized'] for x in results['reverse_results']['tax_values_list'] if x['price_include'])
+
+                        self.assertEqual(
+                            float_round(results['reverse_results']['total_excluded'] + delta, precision_rounding=rounding),
+                            float_round(params['price_unit'], precision_rounding=rounding),
+                        )
+        elif params['test'] == 'adapt_price_unit_to_another_taxes':
+            for results_key in ('py_results', 'js_results'):
+                with self.subTest(results_key=results_key):
+                    if results_key not in test:
+                        continue
+
+                    results = test[results_key]
+                    self.assertEqual(results, test['expected_price_unit'])
+
+    def _assert_tests(self, tests, mode='py_js'):
+        if mode in ('py_js', 'py'):
+            self._add_tests_py_results(tests)
+        if mode in ('py_js', 'js'):
+            self._add_tests_js_results(tests)
+        for i, test in enumerate(tests, start=1):
+            with self.subTest(test_index=i):
```

Comment:
Maybe having a mapping `{sub_test_name: test_vals}` instead of a list/tuple `[test_vals...]` would help having a more meaningful error log message?